### ISSUE TITLE
Problem with custom redirect

### ISSUE THREAD

**Issue by markgerald** [ZF-Commons/ZfcUser#551]
I'm having trouble with the custom redirect.

I am using the code below to redirect:

if (!$auth->hasIdentity() AND ( $matchedRoute == "empresas" )) {
          $redirect = $e->getRouter()->assemble(
                $routeMatch->getParams(),
                array(
                    'name' => $routeMatch->getMatchedRouteName(),
                )
            );

            $response = $e->getResponse();
            $response->getHeaders()->addHeaderLine(
                'Location',
                $e->getRouter()->assemble(
                    array(),
                    array(
                        'name' => 'zfcuser/login',
                        'query' => array('redirect' => $redirect)
                    )
                )
            );
            $response->setStatusCode(302);
            return $response;

But is not working. Can anyone help me? (I'm using the latest version of ZfcUser - 1.2.1), 
PS: This code worked before I upgrade to 1.2.1.

**Comment by markgerald** [comment]
After this change, the redirect parameter passed to the loginAction is totally ignored in the authenticateAction. Anyone can help??

**Comment by pierrejochem** [comment]
I have the same issue :-(

**Comment by jroedel** [comment]
Yep, I think this was a problem introduced very early on in the development cycle. I'm still using 1.0.0 because of this problem. It'd be amazing if someone could take a closer look at this, I'd love to bring myself up-to-date! I'm afraid I don't understand enough about the inner-workings to roll out a pull request...
